File: containers/languages/Dockerfile.lua
FROM mcp-base:latest

USER root

# Install Lua and LuaRocks
RUN apt-get update && apt-get install -y \
    lua5.4 \
    luarocks \
    && rm -rf /var/lib/apt/lists/*

# Install common Lua modules
RUN luarocks install luasocket
RUN luarocks install luajson
RUN luarocks install penlight

USER mcpuser
WORKDIR /workspace

CMD ["lua"]

File: containers/languages/Dockerfile.php
FROM mcp-base:latest

USER root

# Install PHP and common extensions
RUN apt-get update && apt-get install -y \
    php8.1 \
    php8.1-cli \
    php8.1-common \
    php8.1-curl \
    php8.1-json \
    php8.1-mbstring \
    php8.1-xml \
    php8.1-zip \
    composer \
    && rm -rf /var/lib/apt/lists/*

USER mcpuser
WORKDIR /workspace

# Initialize Composer project
RUN composer init --no-interaction

CMD ["php"]
